src/Persistent_cohomology/include/gudhi/Persistent_cohomology.h

```diff
@@ -208,6 +208,44 @@ class Persistent_cohomology {
     }
   }
 
+  /**
+   * @private
+   * Temporary patch to make the computation of more general FilteredComplex possible
+   * Will be removed when a better solution was agreed on.
+   */
+  void compute_persistent_cohomology_without_optimizations(Filtration_value min_interval_length = 0) {
+    if (dim_max_ <= 0)
+      return;  // --------->>
+
+    interval_length_policy.set_length(min_interval_length);
+    Simplex_key idx_fil = -1;
+    std::vector<Simplex_key> vertices; // so we can check the connected components at the end
+    // Compute all finite intervals
+    for (auto sh : cpx_->filtration_simplex_range()) {
+      cpx_->assign_key(sh, ++idx_fil);
+      dsets_.make_set(cpx_->key(sh));
+      int dim_simplex = cpx_->dimension(sh);
+      update_cohomology_groups(sh, dim_simplex);
+    }
+    // Compute infinite intervals of dimension 0
```

Comment:
How come you still have code specific for dimension 0, I thought the point was to handle all dimensions the same?

Reply:
Right, I forgot to remove this loop.